contracts/UniswapV4DeployerCompetition.sol

```diff
@@ -0,0 +1,78 @@
+// SPADIX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.24;
+
+import {Create2} from "openzeppelin-contracts/contracts/utils/Create2.sol";
+import {Owned} from "solmate/auth/Owned.sol";
+import {ERC721} from "solmate/tokens/ERC721.sol";
+import {TokenURILib} from "./libraries/UniswapV4DeployerTokenURILib.sol";
+import {VanityAddressLib} from "./libraries/VanityAddressLib.sol";
+
+contract UniswapV4DeployerCompetition is ERC721 {
+    using VanityAddressLib for address;
+
+    event NewAddressFound(address bestAddress, address minter, uint256 score);
+
+    error InvalidBytecode();
+    error CompetitionNotOver();
+    error CompetitionOver();
+    error NotAllowedToDeploy();
+    error BountyTransferFailed();
+    error WorseAddress();
+
+    bytes32 public bestAddressSalt;
+    address public bestAddress;
+    address public bestAddressSender;
+
+    address public immutable v4Owner;
+    uint256 public immutable competitionDeadline = block.timestamp + 7 days;
+    uint256 public immutable exclusiveDeployDeadline = competitionDeadline + 1 days;
+    bytes32 public immutable initCodeHash;
+
+    constructor(bytes32 _initCodeHash, address _v4Owner) payable ERC721("UniswapV4 Deployer", "V4D") {
```

Comment:
hmmmmmm trying to think through how we can get everything to do the same address on every chain. we would definitely love to not have params here 🤔